src/reducers/photos.js

```diff
@@ -0,0 +1,94 @@
+'use strict'
+
+import {
+  hydratePhotosDataWithAditionalProperites,
+  photosList
+} from '../utility'
+import {
+  DELETE_PHOTOS,
+  TOGGLE_PERMISSION,
+  TOGGLE_SELECT,
+  CLEAR_CHANGE,
+  SET_ORDER,
+  RESET_SELECT_COUNT
+} from '../actions/actionTypes'
+
+const data = hydratePhotosDataWithAditionalProperites(photosList)
+const initialState = {
+  data,
+  changed: false,
+  selectCount: 0,
+  permission: false,
+  orderArray: Object.keys(data)
+}
+
+const photos = (state = initialState, action) => {
+  switch (action.type) {
+    case TOGGLE_SELECT:
+      return {
+        ...state,
+        // increment or decrement selectCount when selection is toggled
+        selectCount: state.data[action.photoId].selected
+          ? state.selectCount - 1
+          : state.selectCount + 1,
```

Comment:
I was just trying to avoid iterating over the whole array every time a toggle was selected or an item was deleted. Could you tell me why iterating over the array makes more sense?